include/ale.h

```diff
@@ -9,35 +9,112 @@
 
 namespace ale {
 
+  /// Interpolation enum for defining different methods of interpolation
   enum interpolation {
+    /// Interpolate using linear interpolation
     linear,
+    /// Interpolate using spline interpolation
     spline
   };
 
-  nlohmann::json constructStateFromIsd(const std::string positionRotationData);
-
+  /**
+   *@brief Get the position of the spacecraft at a given time based on a set of coordinates, and their associated times
+   *@param coords A vector of double vectors of coordinates
```

Comment:
There has to be a way to document the type in a more explicit way. Although, let's get this PR merged in first and worry about it later.